File: app/src/main/java/com/ten/lifecat/phone/activity/BackgroundActivity.java
package com.ten.lifecat.phone.activity;

import android.content.Intent;
import android.os.Bundle;
import android.support.design.widget.NavigationView;
import android.support.v4.widget.DrawerLayout;
import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.widget.Button;
import android.widget.ImageButton;

import com.ten.lifecat.phone.R;

/**
 * 主背景界面
 */
public class BackgroundActivity extends AppCompatActivity {

    private DrawerLayout mDrawerLayout;

    private Button uploadIntentButton;
    private Button webIntentButton;
    private Button photoIntentButton;

    private ImageButton userIntentButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_background);

        regisViews();

        // toolbar
        Toolbar toolbar = findViewById(R.id.background_toolbar);
        setSupportActionBar(toolbar);

        // navigation
        initNavigation();

        // listener
        buttonListener();
    }

    private void initNavigation() {
        NavigationView navigationView = findViewById(R.id.nav_view);

        // 导航按钮
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(true);
            actionBar.setHomeAsUpIndicator(R.mipmap.navigation);
        }

        // 默认激活按钮
        navigationView.setCheckedItem(R.id.nav_profile);

        navigationView.setNavigationItemSelectedListener(item -> {
            Intent intent = new Intent();
            switch (item.getItemId()) {
                case R.id.nav_profile:
                    intent.setClass(BackgroundActivity.this, UserProfileActivity.class);
                    break;
                case R.id.nav_web:
                    intent.setClass(BackgroundActivity.this, WebActivity.class);
                    break;
                case R.id.nav_photo:
                    intent.setClass(BackgroundActivity.this, PhotoActivity.class);
                    break;
                case R.id.nav_upload:
                    intent.setClass(BackgroundActivity.this, UploadActivity.class);
                    break;
                case R.id.nav_settings:
                    intent.setClass(BackgroundActivity.this, SettingsActivity.class);
                    break;
                default:
                    return false;
            }
            startActivity(intent);
            return true;
        });
    }

    private void regisViews() {
        mDrawerLayout = findViewById(R.id.drawer_layout);
        uploadIntentButton = findViewById(R.id.background_button_upload);
        webIntentButton = findViewById(R.id.background_button_web);
        photoIntentButton = findViewById(R.id.background_button_photo);
        userIntentButton = findViewById(R.id.background_button);
    }

    private void buttonListener() {
        uploadIntentButton.setOnClickListener(v -> {
            Intent intent = new Intent();
            intent.setClass(BackgroundActivity.this, UploadActivity.class);
            startActivity(intent);
        });

        webIntentButton.setOnClickListener(v -> {
            Intent intent = new Intent();
            intent.setClass(BackgroundActivity.this, WebActivity.class);
            startActivity(intent);
        });

        photoIntentButton.setOnClickListener(v -> {
            Intent intent = new Intent();
            intent.setClass(BackgroundActivity.this, PhotoActivity.class);
            startActivity(intent);
        });

        userIntentButton.setOnClickListener(v -> {
            Intent intent = new Intent();
            intent.setClass(BackgroundActivity.this, UserProfileActivity.class);
            startActivity(intent);
        });
    }
}

File: app/src/main/java/com/ten/lifecat/phone/view/activity/BackgroundActivity.kt
package com.ten.lifecat.phone.view.activity

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.ImageButton
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import com.google.android.material.navigation.NavigationView
import com.ten.lifecat.phone.R

/**
 * Background main view
 */
class BackgroundActivity : AppCompatActivity() {

    private var mDrawerLayout: androidx.drawerlayout.widget.DrawerLayout? = null

    private var uploadIntentButton: Button? = null
    private var webIntentButton: Button? = null
    private var photoIntentButton: Button? = null

    private var userIntentButton: ImageButton? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_background)

        regisViews()

        // toolbar
        val toolbar = findViewById<Toolbar>(R.id.background_toolbar)
        setSupportActionBar(toolbar)

        // navigation
        initNavigation()

        // listener
        buttonListener()
    }

    private fun initNavigation() {
        val navigationView = findViewById<NavigationView>(R.id.nav_view)

        // 导航按钮
        val actionBar = supportActionBar
        actionBar?.let {
            actionBar.setDisplayHomeAsUpEnabled(true)
            actionBar.setHomeAsUpIndicator(R.mipmap.navigation)
        }

        // 默认激活按钮
        navigationView.setCheckedItem(R.id.nav_profile)
        navigationView.setNavigationItemSelectedListener { item ->
            when (item.itemId) {
                R.id.nav_profile -> startActivity(Intent(this, UserProfileActivity::class.java))
                R.id.nav_web -> startActivity(Intent(this, WebActivity::class.java))
                R.id.nav_photo -> startActivity(Intent(this, PhotoActivity::class.java))
                R.id.nav_upload -> startActivity(Intent(this, UploadActivity::class.java))
                R.id.nav_settings -> startActivity(Intent(this, SettingsActivity::class.java))
            }
            true
        }
    }

    private fun regisViews() {
        mDrawerLayout = findViewById(R.id.drawer_layout)
        uploadIntentButton = findViewById(R.id.background_button_upload)
        webIntentButton = findViewById(R.id.background_button_web)
        photoIntentButton = findViewById(R.id.background_button_photo)
        userIntentButton = findViewById(R.id.background_button)
    }

    private fun buttonListener() {
        uploadIntentButton!!.setOnClickListener { startActivity(Intent(this, UploadActivity::class.java)) }
        webIntentButton!!.setOnClickListener { startActivity(Intent(this, WebActivity::class.java)) }
        photoIntentButton!!.setOnClickListener { startActivity(Intent(this, PhotoActivity::class.java)) }
        userIntentButton!!.setOnClickListener { startActivity(Intent(this, UserProfileActivity::class.java)) }
    }
}
